File: eval-platform/src/main/java/com/sap/hackaton2025/player/service/CsvService.java
package com.sap.hackaton2025.player.service;

import com.sap.hackaton2025.player.model.AircraftType;
import java.io.BufferedReader;
import java.io.FileReader;
import java.util.HashMap;
import java.util.Map;

public class CsvService {

    private static final String DATA_FOLDER = "src/main/resources/liquibase/data/";

    public Map<String, AircraftType> loadAircraftTypes() 
    {
        Map<String, AircraftType> map = new HashMap<>();

        try (BufferedReader br = new BufferedReader(new FileReader(DATA_FOLDER + "aircraft_types.csv")))
         {
            br.readLine(); 
            String line;

            while ((line = br.readLine()) != null) 
                {
                String[] cols = line.split(";");
                AircraftType ac = new AircraftType();
                
                ac.typeCode = cols[1];
                ac.firstClassKitsCapacity = parseInt(cols[7]);
                ac.businessKitsCapacity = parseInt(cols[8]);
                ac.premiumEconomyKitsCapacity = parseInt(cols[9]);
                ac.economyKitsCapacity = parseInt(cols[10]);
                
                map.put(ac.typeCode, ac);
            }
        } catch (Exception e) {
            System.err.println("Eroare incarcare CSV: " + e.getMessage());
        }
        return map;
    }
    private int parseInt(String val) {
        try { return Integer.parseInt(val); } catch (Exception e) { return 0; }
    }
}

File: eval-platform/python_client/player/services/csv_service.py
import os
from player.models import AircraftType

class CsvService:
    DATA_FOLDER = "src/main/resources/liquibase/data/"

    def load_aircraft_types(self):
        aircraft_map = {}
        file_path = os.path.join(self.DATA_FOLDER, "aircraft_types.csv")
        
        if not os.path.exists(file_path):
            print(f"!!! CRITICAL: CSV not found at {file_path}")
            return {}

        try:
            with open(file_path, 'r', encoding='utf-8-sig') as f:
                header = f.readline() # Skip header
                for line in f:
                    line = line.strip()
                    if not line: continue
                    
                    # Split după punct și virgulă
                    cols = line.split(";")
                    
                    if len(cols) < 11:
                        print(f"Skipping invalid line: {line}")
                        continue
                    
                    type_code = cols[1].strip() # Curățăm spațiile
                    
                    try:
                        cap_f = int(cols[7])
                        cap_b = int(cols[8])
                        cap_p = int(cols[9])
                        cap_e = int(cols[10])
                        
                        aircraft_map[type_code] = AircraftType(type_code, cap_f, cap_b, cap_p, cap_e)
                    except ValueError:
                        print(f"Error parsing numbers for {type_code}")

            print(f"   -> [CSV] Loaded {len(aircraft_map)} aircraft types.")
            print(f"   -> [CSV] Available types: {list(aircraft_map.keys())}")
            return aircraft_map
            
        except Exception as e:
            print(f"Error reading CSV: {e}")
            return {}
            
        except Exception as e:
            print(f"Error reading CSV: {e}")
            return {}